account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -0,0 +1,331 @@
+            return date.replace(
+                day=1,
+            )
+
+    @api.multi
+    def _get_statement_date_step(self):
+        self.ensure_one()
+        if self.statement_creation_mode == 'daily':
+            return relativedelta(
+                days=1,
+                hour=0,
+                minute=0,
+                second=0,
+                microsecond=0,
+            )
+        elif self.statement_creation_mode == 'weekly':
+            return relativedelta(
+                weeks=1,
+                weekday=MO,
+                hour=0,
+                minute=0,
+                second=0,
+                microsecond=0,
+            )
+        elif self.statement_creation_mode == 'monthly':
+            return relativedelta(
+                months=1,
+                day=1,
+                hour=0,
+                minute=0,
+                second=0,
+                microsecond=0,
+            )
+
+    @api.multi
+    def _get_statement_date(self, date_since, date_until):
+        self.ensure_one()
+        return date_until.date() - relativedelta(days=1)
+
+    @api.multi
+    def _generate_unique_import_id(self, unique_import_id):
+        self.ensure_one()
+        sanitized_account_number = sanitize_account_number(self.account_number)
+        return (
+            sanitized_account_number and sanitized_account_number + '-' or ''
+        ) + str(self.journal_id.id) + '-' + unique_import_id
+
+    @api.multi
+    def _get_next_run_period(self):
+        self.ensure_one()
+        if self.interval_type == 'minutes':
+            return relativedelta(minutes=self.interval_number)
+        elif self.interval_type == 'hours':
+            return relativedelta(hours=self.interval_number)
+        elif self.interval_type == 'days':
+            return relativedelta(days=self.interval_number)
+        elif self.interval_type == 'weeks':
+            return relativedelta(weeks=self.interval_number)
+
+    @api.model
+    def _scheduled_pull(self):
+        _logger.info('Scheduled pull of online bank statements...')
+
+        providers = self.search([
+            ('active', '=', True),
+            ('next_run', '<=', fields.Datetime.now()),
+        ])
+        if providers:
+            _logger.info('Pulling online bank statements of: %s' % ', '.join(
+                providers.mapped('journal_id.name')
+            ))
+            for provider in providers.with_context({'scheduled': True}):
+                date_since = (
+                    provider.last_successful_run
+                ) if provider.last_successful_run else (
+                    provider.next_run - provider._get_next_run_period()
+                )
+                date_until = provider.next_run
+                provider._pull(date_since, date_until)
+
+        _logger.info('Scheduled pull of online bank statements complete.')
+
+    @api.multi
+    def _obtain_statement_data(
+        self, date_since, date_until
+    ):  # pragma: no cover
```

Comment:
Why the no cover? You can test with the dummy you added.

Reply:
Thanks!